### PR TITLE
Backup as a Service: Backup Policies and Backup Plans 

### PR DESCRIPTION
**Pull request by BShrivastav**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000
https://jiracloud.swg.usma.ibm.com:8443/browse/UI-23392
Output from acceptance testing:

```
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsBackupPoliciesDataSourceBasic'
=== RUN   TestAccIBMIsBackupPoliciesDataSourceBasic
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
--- PASS: TestAccIBMIsBackupPoliciesDataSourceBasic (388.02s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc	389.757s



$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsBackupPolicyPlanDataSourceBasic'
=== RUN   TestAccIBMIsBackupPolicyPlanDataSourceBasic
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
--- PASS: TestAccIBMIsBackupPolicyPlanDataSourceBasic (358.09s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc	359.250s

$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsBackupPolicyPlansDataSourceBasic'
=== RUN   TestAccIBMIsBackupPolicyPlansDataSourceBasic
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
--- PASS: TestAccIBMIsBackupPolicyPlansDataSourceBasic (379.03s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc	380.278s

$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsBackupPolicyDataSourceBasic'
=== RUN   TestAccIBMIsBackupPolicyDataSourceBasic
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
--- PASS: TestAccIBMIsBackupPolicyDataSourceBasic (444.20s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc	445.316s

$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsBackupPolicyPlanBasic'
=== RUN   TestAccIBMIsBackupPolicyPlanBasic
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
--- PASS: TestAccIBMIsBackupPolicyPlanBasic (438.60s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc	439.745s

$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsBackupPolicyBasic'
=== RUN   TestAccIBMIsBackupPolicyBasic
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
--- PASS: TestAccIBMIsBackupPolicyBasic (416.61s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc	417.692s

...
```

### CODE REVIEW

File: examples/ibm-is-ng/main.tf [hunk 1]
```diff
@@ -584,12 +584,14 @@ resource "ibm_is_instance" "instance4" {
 resource "ibm_is_snapshot" "b_snapshot" {
```
Comment (SunithaGudisagarIBM) [inline examples/ibm-is-ng/main.tf:584]: Check  this folder examples->common->github.com/IBM -> there is vpc-go-sdk

File: examples/ibm-is-ng/main.tf [hunk 2]
```diff
@@ -1033,3 +1036,47 @@ data "ibm_is_volumes" "example" {
 data "ibm_is_volumes" "example" {
   zone_name = "us-south-1"
 }
+
+## Backup Policy
+resource "ibm_is_backup_policy" "is_backup_policy" {
```
Comment (SunithaGudisagarIBM) [inline examples/ibm-is-ng/main.tf:1041]: We need to re verify examples here

File: examples/ibm-is-ng/main.tf [hunk 3]
```diff
@@ -1033,3 +1036,47 @@ data "ibm_is_volumes" "example" {
...
+    cron_spec = "30 09 * * *" 
+    deletion_trigger {
+      delete_after      = 20
+      delete_over_count = 20
+    }
+    copy_user_tags = true
+  }
+}
+
+resource "ibm_is_backup_policy_plan" "is_backup_policy_plan" {
```
Comment (SunithaGudisagarIBM) [inline examples/ibm-is-ng/main.tf:1055]: We need to re verify examples here

File: website/docs/d/is_backup_policies.html.markdown [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+subcategory: "VPC infrastructure"
+layout: "ibm"
+page_title: "IBM : is_backup_policies"
+description: |-
+  Get information about backup policies.
+---
+
+# ibm_is_backup_policies
+
+Provides a read-only data source for BackupPolicyCollection. For more information, about backup policy in your IBM Cloud VPC, see [Backup policy](https://cloud.ibm.com/docs/vpc?topic=vpc-creating-backup-policy).
```
Comment (SunithaGudisagarIBM) [inline website/docs/d/is_backup_policies.html.markdown:11]: We need to verify link for all the documents

File: website/docs/d/is_backup_policy.html.markdown [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+```terraform
+provider "ibm" {
+  region = "eu-gb"
+}
+```
+
+## Example Usage
+
+```terraform
+data "ibm_is_backup_policy" "example" {
```
Comment (SunithaGudisagarIBM) [inline website/docs/d/is_backup_policy.html.markdown:27]: Also we need to run terraform fmt on examples here

**Review by SunithaGudisagarIBM — COMMENTED** [review]
Hey Bhavesh, Can you please check review comments and incorporate if they are relevant

File: website/docs/d/is_backup_policies.html.markdown (hunk 4 above)
Comment (BShrivastav) [reply website/docs/d/is_backup_policies.html.markdown]: modified

File: examples/ibm-is-ng/main.tf (hunk 3 above)
Comment (BShrivastav) [reply examples/ibm-is-ng/main.tf:1046]: redundant

File: examples/ibm-is-ng/main.tf (hunk 2 above)
Comment (BShrivastav) [reply examples/ibm-is-ng/main.tf:1041]: done

[the remark this answers is not in the document]
Comment (BShrivastav) [reply examples/ibm-is-ng/main.tf:1041]: verified

File: examples/ibm-is-ng/main.tf (hunk 1 above)
Comment (BShrivastav) [reply examples/ibm-is-ng/main.tf:584]: removed

**Review by SunithaGudisagarIBM — COMMENTED** [review]
Hey, have a look at my PR review comments

File: go.mod [hunk 6]
```diff
@@ -28,7 +28,7 @@ require (
 	github.com/IBM/scc-go-sdk/v3 v3.1.6
 	github.com/IBM/schematics-go-sdk v0.1.3
 	github.com/IBM/secrets-manager-go-sdk v0.1.19
-	github.com/IBM/vpc-go-sdk v0.20.0
+	github.com/IBM/vpc-go-sdk v0.21.0
```
Comment (SunithaGudisagarIBM) [inline go.mod:31]: Change this to new release which is v0.22.0 check in go-sdk release once

**Review by SunithaGudisagarIBM — COMMENTED** [review]
Hey Update Go SDK

File: go.mod (hunk 6 above)
Comment (SunithaGudisagarIBM) [reply go.mod:31]: Also please merge many commits into one commit

**Pull request closed without merging**